### DISCUSSION TITLE
Is there a way to query if any smart queries in the current context are loading or have returned an error?

### DISCUSSION THREAD

**Discussion by toddpadwick** [Akryum/vue-apollo#1242] [Q&A] [answered]
I have a complex search component that has a number of dynamically created apollo smart queries generated in 'for loops' Some of these even have smart queries created inside another parent smart queries update hook as we need the results of one to query the next. 

The final calculation to render the search results, require all data from the smart queries first. This means we cannot use `$apollo.queries.<key>.loading`. but instead need something like `$apollo.queries.loading` or `$apollo.loading` to check for loading smart queries globally.

it would also be useful to have a global way of returning any smart query errors also.

How might I go about doing this?

**Comment by Akryum** [comment, marked as answer]
Use `$apollo.loading` and `$error` in the `apollo` option

```js
apollo: {
  $error (err) { ... }
}
```

**Reply by toddpadwick** [reply]
Ah 🤦‍♂️, I was sure I had tried `$apollo.loading` with no luck, but clearly not. Thanks!
